Add Context to README (#16)
Stedi-Demos/bootstrap: README.md

# Stedi EDI Bootstrap

> This repository contains an end-to-end configuration for building a full X12 EDI system using Stedi products. This implementation demonstrates one way to build an integration for the common read and write EDI use case. Your implementation may differ, depending on your systems and requirements. We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you design a custom integration.

Author: ```suggestion
This repository contains an end-to-end configuration for building a full X12 EDI system using Stedi products. This implementation demonstrates one way to build an integration for the common read and write EDI use case. Your solution may differ, depending on your systems and requirements. We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you design a custom integration.
```

> # Read and Write Workflow 

Author: ```suggestion
## Read and Write Workflow 
```

> The bootstrap uses the following process to handle incoming EDI:

Reviewer: I think we need to come up with a way to refer to the bootstrap repository. I use _bootstrap workflow_ most of the time and only use _bootstrap repository_ if I’m talking about the GitHub repository. I see you use _the bootstrap_. I started out using just _Bootstrap_, but Mohan prefers not treating it like a product, so not capitalization. I’m not saying we have to do it as I did, just that it would be good to pick one way.

> 1. An example trading partner uses [Stedi SFTP](https://www.stedi.com/products/sftp) to send EDI documents to a [Stedi Bucket](https://www.stedi.com/products/buckets).

Reviewer: The product is called Stedi Buckets, the resource is called a Stedi bucket. Which reminds me, I need to add that to the style guide. :-)

> 1. An example trading partner uses [Stedi SFTP](https://www.stedi.com/products/sftp) to send EDI documents to a [Stedi Bucket](https://www.stedi.com/products/buckets).

Reviewer: ```suggestion
1. An example trading partner uses [Stedi SFTP](https://www.stedi.com/products/sftp) to send EDI documents to a [Stedi bucket](https://www.stedi.com/products/buckets).
```

Reviewer: The product is called Stedi Buckets, the resource is called a Stedi bucket. Which reminds me, I need to add that to the style guide. :-)

2. New documents automatically invoke a [Stedi Function](https://www.stedi.com/products/functions) that contains custom code for the workflow. 

> 3. The function calls [Stedi EDI Translate](https://www.stedi.com/products/edi-translate) to transform the EDI data into JSON. This process uses a [Stedi Guide](https://www.stedi.com/products/guides) to map EDI fields to JSON fields. 

Reviewer: I don’t add _Stedi_ to product that start with EDI. I don’t think we decided on a naming scheme, but that’s what I use at the moment. So, Stedi Buckets, but EDI Translate. I started using the _Stedi_ prefix to avoid confusion (since our products have generic names like Buckets, Functions, SFTP, etc.), but I don’t think that’s necessary for names like EDI Translate and EDI Core; no risk of confusion there.

Author: Yeah this is something I keep going back and forth on, so I'm really glad you brought it up. My opinion is that we should always use Stedi in front of the ambiguous ones like you have already been doing, and then be able to use it front of other product names if it's the first reference to the product/if required in context. I realize this may be a point of contention haha, but just saying my thoughts on this. I'm going to look into the way other folks handle this because I want it to be in our editorial style guide. I might leave it in this copy unless you feel VERY strongly about me removing it.

Reviewer: I think I can manage to suppress my outrage over your flaunting of my personal and arbitrary, but clearly superior, convention around naming our products. :wink:

Author: hahaha Joost you're awesome

4. The function sends the JSON document to an internal webhook that you configure.

The bootstrap uses a similar process in reverse to translate JSON documents to EDI and publish them to a bucket.

> # Prerequisites & Deployment

Author: ```suggestion
## Prerequisites & Deployment
```

# Stedi EDI Bootstrap

> This repository contains an end-to-end configuration for building a full X12 EDI system using Stedi products. This implementation demonstrates one way to build an integration for the common read and write EDI use case. Your solution may differ, depending on your systems and requirements. We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you design a custom integration.

Reviewer 2: nit: `s/common read and write EDI use case/common read and write EDI use cases/`

also, I'm not sure about this phrasing: "design a custom integration." maybe something like "configure these workflows to suit your needs" (or something along those lines). cc: @joost-basic-bits for suggestions as well 😄

Reviewer: > nit: s/common read and write EDI use case/common read and write EDI use cases/

Depends on whether you see reading and writing EDI as a single use case or as two.

> I'm not sure about this phrasing: "design a custom integration."

We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you customize the workflow.

Although, what you wrote, works too, Rob. @Laura, now you have options. :smile:

Author: thank you both! :). Design a custom integration is phrasing I think I've seen elsewhere (maybe??) but I really like the suggested edits. I will update to reflect!

Author: also - good call on making use cases plural!

> This repository contains an end-to-end configuration for building a full X12 EDI system using Stedi products. This implementation demonstrates one way to build an integration for the common read and write EDI use case. Your solution may differ, depending on your systems and requirements. We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you design a custom integration.

Author: ```suggestion
This repository contains an end-to-end configuration for building a full X12 EDI system using Stedi products. This implementation demonstrates one way to build an integration for the common read and write EDI use cases. Your solution may differ, depending on your systems and requirements. We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you customize the workflow.
```

## Read and Write Workflow 

The bootstrap workflow uses the following process to handle incoming EDI:

> 1. An example trading partner uses [Stedi SFTP](https://www.stedi.com/products/sftp) to send EDI documents to a [Stedi bucket](https://www.stedi.com/products/buckets).

Reviewer 2: A trading partner _could_ use SFTP (and in practice that's probably the common workflow), but in the instructions / testing section we have the user just upload the file using the Buckets UI, it might be better to indicate that here (could also add that uploads via SFTP will trigger this process as well)

Author: mm okay that's a good point. Let me adjust!

2. New documents automatically invoke a [Stedi Function](https://www.stedi.com/products/functions) that contains custom code for the workflow. 
3. The function calls [Stedi EDI Translate](https://www.stedi.com/products/edi-translate) to transform the EDI data into JSON. This process uses a [Stedi Guide](https://www.stedi.com/products/guides) to map EDI fields to JSON fields. 
4. The function sends the JSON document to an internal webhook that you configure.

> The bootstrap uses a similar process in reverse to translate JSON documents to EDI and publish them to a bucket.

Reviewer 2: nit: it might be good to specifically mention "handle outgoing EDI" here, to align with the description for incoming EDI above.

Author: excellent call - let me adjust accordingly!

> The bootstrap uses a similar process in reverse to translate JSON documents to EDI and publish them to a bucket.

Author: ```suggestion
The bootstrap uses a similar process in reverse to handle outgoing EDI. It translates JSON documents to EDI and publishes them to a Stedi bucket.
```

   npm run bootstrap
   ```

> # Testing the system

Reviewer: Is this correct? This puts _Testing the system_ on the same level as _Stedi EDI Bootstrap_. I think it was correct before.

> # Testing the system

Author: ```suggestion
## Testing the system
```

## Testing the system

> ## Inbound EDI

Author: ```suggestion
### Inbound EDI
```

  </details>

> 5. You can view the file using the [Buckets Web View](https://www.stedi.com/app/buckets). As shown above, the output of the function includes the `bucketName` and `key` (path within the bucket) of where the generated EDI was saved.

Reviewer: ```suggestion
5. You can view the file using the [Buckets UI](https://www.stedi.com/app/buckets). As shown above, the output of the function includes the `bucketName` and `key` (path within the bucket) of where the generated EDI was saved.
```

Reviewer: I know you didn’t change this, Laura, but I missed this before. This is another thing we haven’t been able to agree on, but the docs are pretty consistent these days in using _UI_ for the, well, UI.

Author: thank you!! I am always a fan of opportunistic fixes :)

> # Customizing the Bootstrap

Reviewer: As before: we should decide on how we refer to The Bootstrap.

> # Customizing the Bootstrap

Author: ```suggestion
# Customizing the Bootstrap Workflow
```

The bootstrap uses a sample trading partner to set up and test the read and write EDI workflows. You can customize the bootstrap for your use case by doing one or all of the following:

> - [Adda partner profile](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#add-a-trading-partner-profile) to replace the test trading partner with your real trading partners' details and requirements.

Author: ```suggestion
- [Add a partner profile](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#add-a-trading-partner-profile) to replace the test trading partner with your real trading partners' details and requirements.
```

> - [Add a partner profile](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#add-a-trading-partner-profile) to replace the test trading partner with your real trading partners' details and requirements.

Reviewer 2: nit: there's a contradiction here in `Add` and `replace` (could either be `Edit` and `replace`, or just `Add` and maybe `to represent your real trading...)`

> - [Add a partner profile](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#add-a-trading-partner-profile) to replace the test trading partner with your real trading partners' details and requirements.

Author: ```suggestion
- [Edit the partner profile](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#add-a-trading-partner-profile) to replace the test trading partner with your real trading partners' details and requirements.
```

> - [Create Stedi Mappings](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages) that transform EDI documents into a custom JSON shape for your internal system. The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. 

Reviewer: ```suggestion
- [Create a Stedi mapping](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages) that transform EDI documents into a custom JSON shape for your internal system. The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. 
```

Reviewer: _Stedi Mappings_ is the product. We don’t want our customers to create that; we’d have nothing left to do! ;-)

> - [Create a Stedi mapping](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages) that transform EDI documents into a custom JSON shape for your internal system. The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. 

Reviewer 2: another nit: may want to generalize the use case here, as mapping can be in either direction between guide JSON and a custom API (not necessarily just EDI-like JSON to custom API, could be the other way around too)

Reviewer: > nit: s/common read and write EDI use case/common read and write EDI use cases/

Depends on whether you see reading and writing EDI as a single use case or as two.

> I'm not sure about this phrasing: "design a custom integration."

We encourage you to [contact us](mailto:support@stedi.com) as you get started with Stedi, so we can help you customize the workflow.

Although, what you wrote, works too, Rob. @Laura, now you have options. :smile:

> - [Create a Stedi mapping](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages) that transform EDI documents into a custom JSON shape for your internal system. The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. 

Author: ```suggestion
- [Create a Stedi mapping](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages). The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. You may need to create a mapping to transform EDI documents into a custom JSON shape for your internal system. You can also create a mapping that transforms JSON data from your system into the JSON schema required for outgoing EDI documents.
```

> - [Create a Stedi mapping](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages). The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. You may need to create a mapping to transform EDI documents into a custom JSON shape for your internal system. You can also create a mapping that transforms JSON data from your system into the JSON schema required for outgoing EDI documents.

Author: ```suggestion
- [Create Stedi mappings](https://www.stedi.com/docs/bootstrap/adjusting-the-workflow#map-inbound-messages). The base bootstrap repository ingests and generates JSON with a schema that closely matches EDI documents. You may need to create a mapping to transform EDI documents into a custom JSON shape for your internal system. You can also create a mapping that transforms JSON data from your system into the JSON schema required for outgoing EDI documents.
```
